### DISCUSSION TITLE
How can I map this to the new Remix Router V2 structure?

### DISCUSSION THREAD

**Discussion by superjose** [remix-run/remix#6013] [Q&A]
I have the following folder structure currently in Remix V1

```
admin/
----links/
--------create.tsx
--------index.tsx
----organizations/
--------$organizationId/
-----------------profiles/
--------------------------edit/
-----------------------------------$organizationProfileId.tsx
--------------------------create.tsx
--------------------------index.tsx
--------create.tsx
--------index.tsx
----users/
--------edit/
-----------------$userId.tsx
--------create.tsx
--------create_backup.tsx
--------index.tsx
----root.tsx
api/
----s3/
--------edit.ts
--------index.ts
--------shared.ts
auth/
----reset-password.tsx
----reset.tsx
----sign-in-callback.tsx
cards/
----$cardSlug.tsx
----_index.tsx
cart/
----authenticate.tsx
----checkout.tsx
----_index.tsx
profile/
----links/
--------edit/
-----------------$linkId.tsx
----_index.tsx
view/
----link/
--------$linkId.tsx
--------style.css
----organizations/
--------profiles/
-----------------$organizationProfileId.tsx
----profile/
--------$profileId.tsx
admin.tsx
auth.tsx
healthcheck.tsx
join.tsx
login.tsx
logout.tsx
profile.tsx
_index.tsx
```

![image](https://user-images.githubusercontent.com/8230270/230327146-11a4f171-1974-416d-aeb3-4f48af528b8b.png)

Do you know how I can do this in Remix V2? 

I need clarification when it comes to nesting the folders. 

I've tried this
![image](https://user-images.githubusercontent.com/8230270/230327329-ae93f93a-9401-4fae-a65c-765b6a348fe6.png)

But it won't render correctly. 

I want to group them by folder, as otherwise, it would be a mess. I'd also like to avoid polluting it with `_` underscores wherever it is unnecessary. 

Is there a way?

Thanks 🤗

**Comment by superjose** [comment]
Any ideas? I've reverted to v1 temporarily.

**Comment by MichaelDeBoey** [comment]
Our new ["Preparing for v2" page](https://remix.run/docs/pages/v2#upgrading-to-the-new-convention) could probably help you out

**Reply by Nikohelie** [reply]
Hello

I'm new to remix and I face the same issue.

I try the v2 that keep all the routes to the base folder but with a lot of routes is very confusing.
So I prefer to split the routes with folders convention that match my app use cases.

So in v2 to work with old way we need to work with the workaround package @remix-run/v1-route-convention? 


There are not a cleaner way to do so?

To help you with the answer I create this stackblitz base on your tutorial template 
https://stackblitz.com/edit/remix-run-remix-sxej2u?file=app%2Froot.tsx

Thanks in advance.

**Comment by WesleyYue** [comment]
Fwiw I struggled with organizing file based routing, and realized that you can just [declare the routes in remix.config.js](https://remix.run/docs/en/main/discussion/routes#manual-route-configuration). Migrating to this declared routing approach helped me catch a ton of dead/dangling routes that I didn't intend to expose

IMO this is the simplest, most maintainable way to manage routes. Any changes to the routes on your server would be caught in a PR review.

**Reply by kiliman** [reply]
You might be interested in this then https://github.com/brophdawg11/remix-json-routes

**Reply by WesleyYue** [reply]
Looks interesting! Didn't realize that existed but going to stick with built-in vanilla syntax to keep things as simple as possible. I did find it peculiar that neither of the `remix routes` format can be used in remix.config.js
